Dockerfile

```diff
@@ -15,12 +19,24 @@ RUN addgroup -g 1000 dotnet \
 USER dotnet
 WORKDIR /home/dotnet
 
+# Label images to aid searching
+LABEL uk.gov.defra.dotnetcore.dotnet-version=$NETCORE_VERSION \
+      uk.gov.defra.dotnetcore.version=$DOCKERFILE_VERSION \
+      uk.gov.defra.dotnetcore.repository=defradigital/dotnetcore
+
 # Extend Alpine variant of .Net Core SDK base image for small image size
 FROM mcr.microsoft.com/dotnet/core/sdk:${NETCORE_VERSION}-alpine AS development
```

Comment:
Sort of expected this to be extended from production. If there is a reason to not do that then perhaps a comment?

Reply:
To run an application in .Net Core you need the .Net Core runtime.  The production image is based on Microsoft's .Net Core runtime alpine image.

However, if you want to build a .Net Core application then the runtime alone is not sufficient, you need the .Net Core SDK which is a much more bloated image (several GB if you use the non-alpine version).  So when developing, running tests etc we need to use an image based on the SDK.  If we extended the production image we'd only have the runtime and wouldn't be able to build.

Similarly for production we can't extend the development one because then our production images would be significantly larger than they need to be and have a much greater attack surface for vulnerabilities.

A typical workflow for a .Net Core image with Docker is:
- from SDK image
- build application in container
- from runtime image
- copy built application from SDK image
- run application

We could add more comments to make this clearer but I don't think the purpose of these comments should be to explain how to build and run .Net Core.  I think it's sufficient to say this image is build from the .Net SDK and this one is built from the .Net Runtime.